auction-server/src/state.rs

```diff
@@ -747,15 +747,29 @@ impl Store {
             }
             BidStatus::Lost { result: _, index } => {
                 if let Some(auction) = auction {
-                    query_result = sqlx::query!(
-                        "UPDATE bid SET status = $1, bundle_index = $2, auction_id = $3 WHERE id = $4 AND status = 'submitted'",
-                        updated_status as _,
-                        index.map(|i| i as i32),
-                        auction.id,
-                        bid.id
-                    )
-                    .execute(&self.db)
-                    .await?;
+                    match index {
+                        Some(index) => {
+                            query_result = sqlx::query!(
+                                "UPDATE bid SET status = $1, bundle_index = $2, auction_id = $3 WHERE id = $4 AND status = 'submitted'",
+                                updated_status as _,
+                                index as i32,
+                                auction.id,
+                                bid.id
+                            )
+                            .execute(&self.db)
+                            .await?;
+                        }
+                        None => {
+                            query_result = sqlx::query!(
+                                "UPDATE bid SET status = $1, auction_id = $2 WHERE id = $3 AND status = 'pending'",
+                                updated_status as _,
+                                auction.id,
+                                bid.id
+                            )
+                            .execute(&self.db)
+                            .await?;
+                        }
+                    }
```

Comment:
out of curiosity, is it possible for index to not be None but auction to be None? The query on line 775 seems weird since there is no auction but we are updating the bundle index. Perhaps we should enforce these relationships in another PR